### ISSUE TITLE
Links shared via Share Extension don't have thumbnails or appear in the LINKS section of the Collection view

### ISSUE THREAD

**Issue by AndreaVEH** [wireapp/wire-ios#621]
Thank you, guys. The new Share Extension feature is so great and I turn it on as soon as the app is updated. But I find when I share a link through Share Extension the link just remains strings not a thumbnail as I directly copy the link in the text field and send the message in the app. And the shared links through Share Extension will not appear in the LINKS of the collection view. But the links sent directly in the app will. 
FYI: Version 2.28(2766), iPhone SE 10.2.1

**Comment by AndreaVEH** [comment]
Thank you for your support. I updated Wire to 2.29(2772) and I saw the feature request had been merged. But when I share a link via share extension, the new share window contains the link's web page title followed by the link itself and a image preview appearing on the right. After tapping the share button, both the title and link appear in the conversation timeline in the format of strings, the small image preview doesn't appear either. And the shared links via share extension still not appear in the LINKS of the collection view. The feature request is not included in this update, is it?

**Comment by jenskasemets** [comment]
Hi @AndreaVEH , this issue is now fixed in release 2.30, please let us know if you still have issues with it.

**Comment by AndreaVEH** [comment]
@jenskasemets , I find the issue still exists when the link is http protocol. But when the link is https protocol, the issue doesn't exist. And if I directly paste http or https links into the text input field, everything works well. My current version is Version 2.33(2811).

**Comment by AndreaVEH** [comment]
@jenskasemets @KoalaBoomka , I also find Youtube links, both https://www.youtube.com/* and https://www.youtu.be/*, will have previews, but will not be added to the LINKS or VIDEOS of the Collection view whether I use share extension to share or directly paste links in the input text field.

**Comment by daehn** [comment]
Hello @AndreaVEH, thanks for getting in touch. We recently discovered that link previews were only generated for websites encoded with a UTF-8 charset, but this issue will be fixed in an upcoming release. Besides this issue there are two other things to mention about the generation of link previews:

* We can only generate a rich link preview if the source site contains [Open Graph](http://ogp.me) markup, if this is not the case we still try to use the website's title though.
* You are right that YouTube (and a small selection of other sites) links are currently not added to the links or video section, as we still use an older version of our link preview generation for these sites. For these sites we do not use the Open Graph data yet and are unable to show them in the Collection view at this point. I would recommend opening a separate issue for this to track the progress of this issue there, as it is unrelated to sending rich lins from the iOS Share Extension and more a general issue of the handling of YouTube links.

**Comment by AndreaVEH** [comment]
Hello @daehn , thanks for reply. What I mean in the first one is the same http link that shared via share extension would not get a preview, but directly pasted in the input text field and sent would.

![fullsizerender](https://cloud.githubusercontent.com/assets/7317368/25538258/96af32d0-2c74-11e7-973b-f1baa2b5d2c1.jpg)

Like the screenshot above, red rectangle is shared via share extension and the blue one is pasted and sent via input text field directly. They are the same link, but not https link.

**Comment by AndreaVEH** [comment]
Hello @daehn , yesterday The Verge announced it was HTTPS by default. So I try to share the link in the screenshot again via share extension. Because the link is https protocol, the link has a preview and is listed in the LINKS of Collection view as expected this time.

**Comment by AndreaVEH** [comment]
Hello @daehn , I've updated Wire to Version 2.35(2825). The issue that sharing http links via share extension wouldn't have a preview or be listed in the LINKS of Collection view, but pasting the same http links into the text input field and sending directly would works well as expected still exists.

**Comment by AndreaVEH** [comment]
Hello @daehn , would you make this issue reopen as there is something wrong with the http links sharing through share extension?

**Comment by AndreaVEH** [comment]
Hello @daehn , in the latest release notes of iOS 2.36, I saw "Sharing links from other apps would sometimes get stuck and never send the link." in the fixed section.  So I tried to test the problem mentioned in this issue.  But the problem still seems to exist. 

![image](https://cloud.githubusercontent.com/assets/7317368/26518034/6cb767ec-42d9-11e7-82b5-fc88effdb992.png)

I screenshot the problem.  The first http link with a preview is directly pasted into the text input field and sent.  The second one is shared via Share Extension from Safari.
